src/libstd/sys/windows/process.rs

```diff
@@ -141,12 +137,33 @@ impl Command {
                                    .with_extension(env::consts::EXE_EXTENSION);
                     if fs::metadata(&path).is_ok() {
                         return Some(path.into_os_string())
+                    } else {
+                        // Windows relies on path extensions to resolve commands.
+                        // Path extensions are found in the PATHEXT environment variable.
+                        if let Some(exts) = self.env.get("PATHEXT") {
+                            for ext in split_paths(&exts) {
+                                let ext_str = pathext.to_str().unwrap().trim_matches('.');
+                                let path = path.join(self.program.to_str().unwrap())
```

Comment:
Here I think you just need to call `with_extension` as otherwise this is joining `self.program` onto the root path twice I think.